modules/300-prometheus/monitoring/prometheus-rules/prometheus-storage.yaml

```diff
@@ -0,0 +1,52 @@
+- name: d8.prometheus.storage
+  rules:
+  - alert: PrometheusDiskUsage
+    expr: |
+      d8_prometheus_fs_used_percent > 95
+    labels:
+      tier: cluster
+      severity_level: "4"
+    annotations:
+      plk_markup_format: markdown
+      plk_protocol_version: "1"
+      plk_pending_until_firing_for: "10m"
+      summary: Prometheus disk is over 95% used.
+      description: |
+        For more information, use the command:
+        ```shell
+        kubectl -n {{ $labels.namespace }} exec -ti {{ $labels.pod_name }} -c prometheus -- df -PBG /prometheus
+        ```
+        Consider increasing it https://deckhouse.io/en/documentation/v1/modules/300-prometheus/faq.html#how-to-expand-disk-size
+  # If there was retention in the last 24 hours and the oldest metric was recorded less than 10 days ago and storage.tsdb.retention.size has not been changed.
+  - alert: PrometheusMainRetainLessThan10Days
+    expr: |
+      group by (service) (increase(prometheus_tsdb_size_retentions_total{service="prometheus"}[24h]) > 0)
```

Comment:
I didn't get the idea of the query. I think it should be enough just to use the second part.
```
(time() - prometheus_tsdb_lowest_timestamp_seconds{service="prometheus"}) / 86400 < 10
```
@vitaliy-sn could you please elaborate more on why two additional checks are required?

Reply:
The first expression excludes false positives when the module is just turned on.
The last expression turns off the alert if the disk has been enlarged.